assets/javascripts/discourse/lib/add-recurrent-events.js

```diff
@@ -1,18 +1,30 @@
 import DiscoursePostEventEvent from "../models/discourse-post-event-event";
 
-export default function addRecurrentEvents(events) {
-  return events.flatMap((event) => {
-    const upcomingEvents =
-      event.upcomingDates?.map((upcomingDate) =>
-        DiscoursePostEventEvent.create({
-          name: event.name,
-          post: event.post,
-          category_id: event.categoryId,
-          starts_at: upcomingDate.starts_at,
-          ends_at: upcomingDate.ends_at,
-        })
-      ) || [];
+export default async function addRecurrentEvents(eventsPromise) {
+  try {
+    const events = await eventsPromise;
```

Comment:
I think it's a weird to do the `await` in here.. It feels like this function should expect an array instead of a promise